supervised/embeddings.py

```
@@ -0,0 +1,73 @@
+""" Processing embeddings in PyTorch"""
+
+from __future__ import print_function, unicode_literals, division
+
+import gensim
+import torch
+import torch.nn as nn
+from torchnlp.word_to_vector import GloVe
+import os
+
+LOAD_FUNCS = {'word2vec': load_word2vec, 
+			  'glove': load_GloVe, 
+			  'elmo': load_ElMo,
+			  'bert': load_BERT}
+
+def load_word2vec(path_to_file, freeze=False, binary=True):
+	"""Loads the word2vec embeddings"""
+	model = gensim.models.KeyedVectors.load_word2vec_format(path_to_file, binary=binary)
+	embedding_vectors = torch.FloatTensor(model.vectors)
+	word_to_index = {word: vector.index for word, vector in model.vocab.items()}
+	embedding = nn.Embedding.from_pretrained(embedding_vectors, freeze=freeze)
+	return embedding, word_to_index
+
+def load_GloVe(path_to_file, freeze=False, binary=True):
+	"""Loads pretrained GloVe embeddings"""
+	model_dict = GloVe(name='840B', dim=300)
+	embedding_vectors = list(model_dict.values())
+	embedding_vectors = Torch.FloatTensor(embedding_vectors)
+	word_to_index = {word: i for i, word in enumerate(list(model_dict.keys()))}
+	embedding = nn.Embedding.from_pretrained(embedding_vectors, freeze=freeze)
+	return vectors, word_to_index
+
+def load_ElMo(path_to_file, freeze=False, binary=True):
+	raise NotImplementedError()
+
+def load_BERT(path_to_file, freeze=False, binary=True):
+	raise NotImplementedError()
+
+def load_pretrained(path_to_file, embed_type='word2vec', freeze=False, binary=True):
+	"""Loads pretrained embeddings and returns embeddings and word to index"""
+	if not os.path.exists(path_to_file):
+		raise FileNotFoundError('File does not exist. Please download embeddings and try again.')
+
+	try:
+		load_func = LOAD_FUNCS[embed_type]
+		return load_func(path_to_file, freeze=freeze, binary=binary)
+
+	except KeyError:
```

Remove this and just let it fail. Also, we don't need LOAD_FUNCS at all, see above. It's experimental code, after all.